packages/PostRector/Rector/ClassRenamingPostRector.php

```diff
@@ -51,7 +53,9 @@ public function enterNode(Node $node): ?Node
             return null;
         }
 
-        $result = $this->classRenamer->renameNode($node, $oldToNewClasses);
+        /** @var Scope|null $scope */
+        $scope = $node->getAttribute(AttributeKey::SCOPE);
```

Comment:
remove `$node->getAttribute(AttributeKey::SCOPE)` deep in the service, but I had to add it here, as I don't know a way on how to get the `Scope` in a `AbstractPostRector`.

Its good enough for now and we can revisit later, in case there is no way to handle that atm

Reply:
Moving scope pull here is possibly not an improvement imo.

Scope is pulled from origNode on refactor()

PostRector means the scope already refreshed, so it may cause unexpected result.

Reply:
ok, so you say that we should pass in the Scope into the post-rector from the caller side?

Reply:
I think scope need to be pulled from origNode when possible make consistent:

https://github.com/rectorphp/rector-src/blob/3f8cc390d07486978300abe3dea9f36eeb017c5a/src/Rector/AbstractScopeAwareRector.php#L32-L36

That same with in AbstractRector as well:

https://github.com/rectorphp/rector-src/blob/3f8cc390d07486978300abe3dea9f36eeb017c5a/src/Rector/AbstractRector.php#L342-L343

I tried to use Node instead of origNode and fail on renaming.